app/src/main/java/org/fossasia/openevent/app/module/event/create/CreateEventPresenter.java

```diff
@@ -53,11 +58,100 @@ private boolean verify() {
         return true;
     }
 
+    private boolean findEmptyLink(String uri) {
+        if (uri == null)
+            return false;
+        else return !("".equals(uri));
+    }
+
+    private void resolveEmptyLinks() {
+
+        if (!findEmptyLink(event.getLogoUrl())) {
+            event.setLogoUrl(null);
+        }
+        if (!findEmptyLink(event.getTicketUrl())) {
+            event.setTicketUrl(null);
+        }
+        if (!findEmptyLink(event.getOriginalImageUrl())) {
+            event.setOriginalImageUrl(null);
+        }
+        if (!findEmptyLink(event.getExternalEventUrl())) {
+            event.setExternalEventUrl(null);
+        }
+        if (!findEmptyLink(event.getPaypalEmail())) {
+            event.setPaypalEmail(null);
+        }
+    }
+
+    public void validateUrl(TextInputLayout textInputLayout) {
+
+        textInputLayout.getEditText().addTextChangedListener(new TextWatcher() {
+
+            @Override
+            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+                // Nothing here
+            }
+
+            @Override
+            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+                if (ValidateUtils.validateUrl(charSequence.toString())) {
+                    textInputLayout.setError(null);
+                    textInputLayout.setErrorEnabled(false);
+                } else {
+                    textInputLayout.setErrorEnabled(true);
+                    textInputLayout.setError("Please Enter a Valid Url");
+                }
+                if (TextUtils.isEmpty(charSequence)) {
+                    textInputLayout.setError(null);
+                    textInputLayout.setErrorEnabled(false);
+                }
+            }
+
+            @Override
+            public void afterTextChanged(Editable editable) {
+                // Nothing here
+            }
+        });
+    }
```

Comment:
If we can't unit test something in Presenter, it's of no use. This should be present in View